sections/docker.zsh

```diff
@@ -46,22 +46,28 @@ spaceship_docker() {
     [[ "$compose_exists" == false ]] && return
   fi
 
-  # Show Docker status only for Docker-specific folders
-  [[ "$compose_exists" == true || -f Dockerfile || -f docker-compose.yml || -f /.dockerenv ]] || return
+  # Show Docker status only for Docker-specific folders or when connected to external host
+  [[ "$compose_exists" == true || -f Dockerfile || -f docker-compose.yml || -f /.dockerenv || -n $DOCKER_MACHINE_NAME || -n $DOCKER_HOST ]] || return
```

Comment:
Yes, I know it behaves differently in this aspect, but I did so on purpose. Every single docker command you issue will be executed on the DOCKER_HOST, no matter what directory you're in.

So it isn't really comparable to spaceship's prompt extensions for directory based programming contexts, where code is only executed in the current directory. Instead, it behaves more like working over `ssh`. And spaceship (like so many other prompts) always shows that you're currently working on a remote machine in your prompt (if it's installed there as well). **Very** different result if you execute a `rm -rf` by accident on a remote server or your local machine. ;)

Reply:
If there is a need to change how a spaceship-prompt plugin behaves, the following approach might be useful. In `.zshrc`:
```bash
# create a new function for your custom prompt section
spaceship_kubeconfig() {
  # exit on desired condition
  [[ -z $KUBECONFIG ]] && return

  # run the stock section
  spaceship_kubecontext
}

# use the custom section instead of a stock one
SPACESHIP_RPROMPT_ORDER=( ... kubeconfig ... )
```